src/main/scala/com/adobe/api/platform/runtime/metrics/PrometheusRecorder.scala

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+ */
+
+package com.adobe.api.platform.runtime.metrics
+import java.io.StringWriter
+import java.util
+import java.util.concurrent.TimeUnit
+
+import akka.http.scaladsl.model.{HttpEntity, MessageEntity}
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.adobe.api.platform.runtime.metrics.Activation.getNamespaceAndActionName
+import com.adobe.api.platform.runtime.metrics.MetricNames._
+import io.prometheus.client.exporter.common.TextFormat
+import io.prometheus.client.{CollectorRegistry, Counter, Histogram}
+
+import scala.collection.JavaConverters._
+import scala.collection.concurrent.TrieMap
+
+object PrometheusRecorder extends MetricRecorder with PrometheusExporter {
+  private val metrics = new TrieMap[String, PrometheusMetrics]
+  private val activationCounter = counter(activationMetric, "Activation Count", "namespace", "action")
+  private val coldStartCounter = counter(coldStartMetric, "Cold start counts", "namespace", "action")
+  private val statusCounter = counter(statusMetric, "Activation failure status type", "namespace", "action", "status")
+  private val waitTimeHisto = histogram(waitTimeMetric, "Internal system hold time", "namespace", "action")
+  private val initTimeHisto =
+    histogram(initTimeMetric, "Time it took to initialize an action, e.g. docker init", "namespace", "action")
+  private val durationHisto =
+    histogram(durationMetric, "Actual time the action code was running", "namespace", "action")
+
+  private val metricSource = createSource()
+
+  def processEvent(activation: Activation): Unit = {
+    lookup(activation.name).record(activation)
+  }
+
+  override def getReport(): MessageEntity =
+    HttpEntity(PrometheusExporter.textV4, metricSource)
+
+  private def lookup(name: String): PrometheusMetrics = {
+    //TODO Unregister unused actions
+    metrics.getOrElseUpdate(name, {
+      val (namespace, action) = getNamespaceAndActionName(name)
+      PrometheusMetrics(namespace, action)
+    })
+  }
+
+  case class PrometheusMetrics(namespace: String, action: String) {
```

Comment:
For above usecases its better to have them captured at global level (without the action specific dimension). 

* Memory - Have a new histogram with custom buckets
* Kind - Metric with `kind` as one of the dimension

As said this can be done in a separate PR